### PR TITLE
Generate cross section information in (py)hepmc files

### PR DESCRIPTION
**Pull request by afedynitch**
Add inelastic cross section to hepmc output.

Since chromo generates only successful events, the generated cross section is equal to the estimated one.

### CODE REVIEW

File: src/chromo/models/qgsjet.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def _cross_section(self, kin=None, full_AA=False):
         """
         kin = self.kinematics if kin is None else kin
 
-        if full_AA and (kin.p1.is_nucleus and kin.p1.A > 1):
+        if max_info and (kin.p1.is_nucleus and kin.p1.A > 1):
```
Comment (jncots) [inline src/chromo/models/qgsjet.py:142]: Could target be anything (kin.p2)?

File: src/chromo/models/qgsjet.py [hunk 2]
```diff
@@ -262,7 +262,7 @@ def _cross_section(self, kin=None, full_AA=False):
...
```
Comment (jncots) [inline src/chromo/models/qgsjet.py:265]: Could kin.p2 be anything?

**Review by jncots — COMMENTED** [review]
Phew, that's a lot of changes for a small modification. I've noted some unclear or suspicious places.

File: src/chromo/models/qgsjet.py (hunk 1 above)
Comment (afedynitch) [reply src/chromo/models/qgsjet.py:142]: Usually the possible targets are restricted by whatever is listed in `_targets`. So there is no explicit need to check this. This function is the correct one to call if the target is heavier than a nucleon or if the projectile is a nucleus. Save for the case below and for dpmjet above.

Comment (afedynitch) [reply src/chromo/models/qgsjet.py:142]: But yeah, actually, there is no check if an externally constructed kinematics object is (without calling `model.kinematics = kin`) is passed

File: src/chromo/common.py [hunk 3]
```diff
@@ -196,6 +196,7 @@ class EventData:
     nevent: int
     impact_parameter: float
     n_wounded: Tuple[int, int]
+    production_cross_section: float
```
[the remark this answers is not in the document]
Comment (afedynitch) [reply src/chromo/common.py:199]: fixed

File: src/chromo/models/dpmjetIII.py [hunk 4]
```diff
@@ -142,12 +142,11 @@ def __init__(self, evt_kin, *, seed=None):
 
         self._set_final_state_particles()
 
-    def _cross_section(self, kin=None, photon_x=0):
+    def _cross_section(self, kin=None, photon_x=0, max_info=False):
```
[the remark this answers is not in the document]
Comment (afedynitch) [reply src/chromo/models/dpmjetIII.py:145]: I'm not sure what photon_x means. I guess it is Q^2/s, but not 100% convinced. Anyhow, it's a private parameter and one can call it by calling the hidden `_cross_section` function.

File: src/chromo/models/dpmjetIII.py [hunk 5]
```diff
@@ -184,6 +183,12 @@ def _generate():
                 + glxs.xsqe2[0, 0, 0]
                 + glxs.xsela[0, 0, 0],
             )
+        elif (kin.p1.is_nucleus and kin.p1.A > 1) or kin.p2.A > 1:
```
[the remark this answers is not in the document]
Comment (afedynitch) [reply src/chromo/models/dpmjetIII.py:186]: ok

File: src/chromo/models/qgsjet.py (hunk 2 above)
Comment (afedynitch) [reply src/chromo/models/qgsjet.py:265]: yes. Everything from _targets

**Comment by afedynitch** [comment]
Thanks! Fixes implemented

**Review by jncots — APPROVED** [review]
Ready to merge

**Pull request merged**